src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -258,6 +245,49 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 			}
 		}
 
+		private void UpdateEntriesFromInputApk (string apkInputPath, ZipArchiveEx apk, DateTime lastWriteOutput, DateTime lastWriteInput, HashSet<ulong> deletedEntries)
+		{
+			using (var packaged = new ZipArchiveEx (apkInputPath, FileMode.Open)) {
+				foreach (var entry in packaged.Archive) {
+					Log.LogDebugMessage ($"Deregistering item {entry.FullName}");
+					existingEntries.Remove (entry.FullName);
+					if (lastWriteInput <= lastWriteOutput)
+						continue;
+
+					long entryIndexInOutput;
+					if (apk.Archive.ContainsEntry (entry.FullName, out entryIndexInOutput)) {
+						ZipEntry e = apk.Archive.ReadEntry (entry.FullName);
+						// check the CRC values as the ModifiedDate is always 01/01/1980 in the aapt generated file.
+						if (entry.CRC == e.CRC) {
```

Comment:
@grendello: what's the "likelihood" of CRC collisions?  This is "only" a 32-bit value.  Should we be relying on it for change "identicality" checks?

Reply:
For ZIP purposes it's not likely that we'd have collisions. Adler-32 (used to calculate the CRC) is weak for short messages, however, so there is a possibility of collision on small files. How big a likelihood? Depends on the data set... I wouldn't use it alone to see if the data has changed or not, but combined with the file name check I think it's good enough.